contracts/src/v0.8/ccip/pools/GHO/UpgradeableTokenPool.sol

```diff
@@ -317,4 +323,28 @@ abstract contract UpgradeableTokenPool is IPool, OwnerIsCreator, IERC165 {
     if (IARM(i_armProxy).isCursed()) revert BadARMSignal();
     _;
   }
+
+  /// @notice Getter for legacy onRamp address.
+  /// @param remoteChainSelector The remote chain selector for which the legacy onRamp is being retrieved.
+  /// @return legacyOnRamp The legacy onRamp address for the given remoteChainSelector
+  function getLegacyOnRamp(uint64 remoteChainSelector) public view returns (address legacyOnRamp) {
+    assembly ("memory-safe") {
+      mstore(0, LEGACY_ON_RAMP_STORAGE_SLOT)
+      mstore(32, remoteChainSelector)
+      legacyOnRamp := shr(96, shl(96, sload(keccak256(0, 64))))
+    }
+  }
+
+  /// @notice Setter for legacy onRamp address, only callable by the DAO.
+  /// @param remoteChainSelector The remote chain selector for which the legacy onRamp is being set.
+  /// @param legacyOnRamp The address of the legacy onRamp.
+  function setLegacyOnRamp(uint64 remoteChainSelector, address legacyOnRamp) external onlyOwner {
+    if (!isSupportedChain(remoteChainSelector)) revert ChainNotAllowed(remoteChainSelector);
+    // todo: add check for typeAndVersion or getStaticConfig().destChainSelector on legacyOnRamp?
```

Comment:
Resolve or remove this todo?

Reply:
made it immutable - [6278e91](https://github.com/aave/ccip/pull/18/commits/6278e91bf61578dae457abb37e4107de1ef5a341)